cmd/kube-controller-manager/app/core.go

```diff
@@ -192,9 +193,14 @@ func startAttachDetachController(ctx ControllerContext) (http.Handler, bool, err
 	if ctx.ComponentConfig.AttachDetachController.ReconcilerSyncLoopPeriod.Duration < time.Second {
 		return nil, true, fmt.Errorf("Duration time must be greater than one second as set via command line option reconcile-sync-loop-period.")
 	}
+	csiClientConfig := ctx.ClientBuilder.ConfigOrDie("attachdetach-controller")
+	// csiClient works with CRDs that support json only
+	csiClientConfig.ContentType = "application/json"
```

Comment:
This shouldn't have been necessary.  Clients should have priority order content accept types which should result in json being served.  What happened when you didn't set this?

Reply:
@liggitt wouldn't have expected this to show up in code using CRD

Reply:
> What happened when you didn't set this?

`Watch()` did not work. It received protobuf and failed to parse it.

Reply:
is this related to https://github.com/kubernetes/kubernetes/pull/62175?